src/drivers/opfs.ts

```diff
@@ -0,0 +1,109 @@
+import { defineDriver } from "./utils";
+import {
+  DRIVER_NAME,
+  exists,
+  getFileObject,
+  joinPaths,
+  normalizePath,
+  readFile,
+  readdirRecursive,
+  remove,
+  removeChildren,
+  unlink,
+  writeFile,
+} from "./utils/opfs-utils";
+
+export interface OPFSStorageOptions {
+  /**
+   * The filesystem root to use
+   * Defaults to the OPFS root at `navigator.storage.getDirectory()`
+   */
+  fs?: FileSystemDirectoryHandle | Promise<FileSystemDirectoryHandle>;
+
+  /**
+   * The base path to use for all operations
+   * Defaults to the root directory (empty string)
+   */
+  base?: string;
+
+  /**
+   * A callback to ignore certain files in getKeys()
+   */
+  ignore?: (path: string) => boolean;
+
+  /**
+   * Whether to ignore any write operations
+   */
+  readOnly?: boolean;
+
+  /**
+   * Whether to disallow clearing the storage
+   */
+  noClear?: boolean;
+}
+
+export default defineDriver<OPFSStorageOptions | undefined>(
+  (opts: OPFSStorageOptions = {}) => {
+    const fsPromise = Promise.resolve(
+      opts.fs ?? navigator.storage.getDirectory()
```

Comment:
We might have a runtime protection to see if `navigator?.storage` is available and otherwise throw an explicit error.

Also this needs to be via a function like `await getFS()`. Unstorage drivers are expected to be lazy initialized.

Reply:
Alright, done.